### PR TITLE
Pinpp update to pin3.14 for x64 linux using make

### PR DESCRIPTION
**Pull request by Kirbk**
builds pin tools using pinpp on pin3.14, cpp11 still doesn't work in pin though so a lot of functionality is lacking.

### CODE REVIEW

File: config/pin_base.mpb [hunk 1]
```diff
@@ -34,7 +36,7 @@ project {
   }
 
   specific (gnuace, make) {
-    compile_flags += -O3 -fomit-frame-pointer -fno-strict-aliasing -Wno-unknown-pragmas -fno-stack-protector -fPIC
+    compile_flags += -O3 -funwind-tables -fno-stack-protector -fasynchronous-unwind-tables -fomit-frame-pointer -fno-strict-aliasing -fno-rtti -faligned-new -fPIC -isystem $(PIN_ROOT)/extras/stlport/include -isystem $(PIN_ROOT)/extras/libstdc++/include -isystem $(PIN_ROOT)/extras/crt/include -isystem $(PIN_ROOT)/extras/crt/include/arch-x86_64 -isystem $(PIN_ROOT)/extras/crt/include/kernel/uapi -isystem $(PIN_ROOT)/extras/crt/include/kernel/asm-x86_64 -I$(PIN_ROOT)/source/include/pin -I$(PIN_ROOT)/source/include/pin/gen -I$(PIN_ROOT)/extras/components/include -I$(PIN_ROOT)/extras/xed-intel64/include/xed -nostdlib -fpermissive -Wno-error=all -fno-exceptions
```
Comment (jwillemsen) [inline config/pin_base.mpb:39]: You add directories which look to be already in the intel64 feature, see https://github.com/DOCGroup/MPC/blob/b3f6d22dcb39d125475651b580e9a5805767b6a1/config/pin_base.mpb#L91. The generic part of pin_base shouldn't have architecture specifc includes

File: config/pintool.mpb [hunk 2]
```diff
@@ -14,11 +14,11 @@ project : pin {
...
     ifeq ($(shell uname -s), Darwin)
       CPPFLAGS += -DTARGET_MAC
-      LDFLAGS  += -lpindwarf -lpinpthread -Wl,-exported_symbols_list -Wl,$(PIN_ROOT)/source/include/pin/pintool.exp
+      LDFLAGS  += $(PIN_ROOT)/intel64/runtime/pincrt/crtbeginS.o -nostdlib -ldl-dynamic -lunwind-dynamic -lpin3dwarf -lpinpthread -Wl,-exported_symbols_list -Wl,$(PIN_ROOT)/source/include/pin/pintool.exp
       LDFLAGS  -= -lpthread
     else
-      CPPFLAGS += -DTARGET_LINUX -Wl,--hash-style=sysv
-      LDLIBS   += -lpindwarf
+      CPPFLAGS += -DTARGET_LINUX -fno-rtti -Wl,--hash-style=sysv $(PIN_ROOT)/intel64/runtime/pincrt/crtendS.o
+      LDLIBS   += -nostdlib -ldl-dynamic -lunwind-dynamic -lpin3dwarf -L$(PIN_ROOT)/intel64/runtime/pincrt
```
Comment (jwillemsen) [inline config/pintool.mpb:21]: Same here, doesn't look correct to have intel64 in a generic part

**Review by jwillemsen — CHANGES_REQUESTED** [review]
I don't have any experience with pin but the proposed changes look to add intel64 specific flags in the generic part, see the intel64 and ia32 features that the pin*.mpb base projects have

File: config/pin_base.mpb (hunk 1 above)
Comment (Kirbk) [reply config/pin_base.mpb]: moved into feature and took out duplicates.

File: config/pintool.mpb (hunk 2 above)
Comment (Kirbk) [reply config/pintool.mpb]: moved out of generic part.

**Pull request merged**